Make the editor responsive

jasmussen opened the pull request:
This PR makes the editor responsive:

- It makes the body responsive
- It hides the up/down arrows at the smallest mobile breakpoint, according to the mockups we'll want to show these in top toolbars at that breakpoint regardless, which I will ticket separately
- It makes the sidebar responsive
- The sidebar, when toggled open, covers the entire viewport at the mobile breakpoint, convenient that you can just toggle it off

jasmussen committed:
Make sidebar Y-position responsive.

jasmussen committed:
Consider up/down arrows, and make sidebar smaller.

This commit shows up/down arrows only on larger breakpoints. The sidebar also becomes smaller at lower breakpoints.

jasmussen committed:
Futher sidebar responsiveness tweaks.

Resize at proper breakpoints.

Hide at mobile. More to come.

Fix regression in toggled tab buttons.

jasmussen committed:
Make sidebar responsive to mobile.

When on mobile, the sidebar covers the whole screen. You have to toggle it off.

aduth commented:
Full-width image styles from #518 will need to be updated in a rebase, which should go to hint at the agony we'll experience with a mid-range sidebar width. 😬

aduth commented on editor/layout/style.scss:
```diff
@@ -1,3 +1,22 @@
+}
+.editor-layout.is-sidebar-opened .editor-sidebar {
+	/* Sidebar covers screen on mobile */
+	width: 100%;
```
Should we apply these styles in `sidebar/style.scss` as:

```
.editor-sidebar {

	.editor-layout.is-sidebar-opened & {
		// ...
	}
}
```

Or if not we should probably update text editor toolbar for consistency:

https://github.com/WordPress/gutenberg/blob/44c915b/editor/modes/text-editor/style.scss#L40-L42

Seems to be deciding between letting the component be responsible for its own style but make assumptions about existence of ancestor classes, vs. applying them from the ancestor at the cost of losing colocation between the component and all its styles.

jasmussen committed:
Reduce to one sidebar width.

Also move sidebar responsive styles from layout to the component.

jasmussen committed:
Reorganize CSS further to sidebar component.

jasmussen commented:
Rebased, and did a few changes based on your great feedback:

- Reduced to a single sidebar width, 280px, which incidentally is the same size as currently in the editor. Might as well start simple and add complexity later
- Moved the CSS for responsive sidebar all into the sidebar component, which makes sense to me. 
- Made sure the text editor mode worked well with the sidebar.

The fixed header in the text editor mode will need love separately. However I'm going to open a ticket soon about "undocking toolbars" —  Calypso style. I'm crossing my fingers we can find a good way to make fixed position toolbars, perhaps we need to use JS for this to avoid the headaches. In other words I'd like for us to tackle improvements to fixed positioning separately. 

Good for a final review?

youknowriad commented on editor/components/button/style.scss:
```diff
@@ -7,14 +7,15 @@
-		background: $dark-gray-500;
-		color: $white;
-	}
+	.editor-tools__tabs & {
```
Minor: We already have some button styles in the tools stylesheet. Should we move these styles there to keep the Button component generic?

youknowriad reviewed (approved):
LGTM 👍
